DesignPathSearch_II/Program.cs

```
@@ -0,0 +1,79 @@
+        public static readonly string connDestination = ConfigurationManager.AppSettings["DestinationConnection"].ToString();
+
+        public static readonly string XtplDir = ConfigurationManager.AppSettings["XtplDir"].ToString();
+
+        static void Main(string[] args)
+        {
+            // output stuff
+            string reportPath = ConfigurationManager.AppSettings["ReportDir"].ToString();
+            string reportName = ConfigurationManager.AppSettings["ReportName"].ToString();
+            string date = DateTime.Now.ToString("yyyy-MM-dd-HHmm");
+            string reportFile = Path.Combine(reportPath, reportName + "-" + date + ".txt");
+            bool isInsert2DB = Boolean.Parse(ConfigurationManager.AppSettings["InsertIn2DB"] ?? "false");
+
+            if (File.Exists(reportFile))
+                File.Delete(reportFile);
+            StreamWriter file = new StreamWriter(reportFile);
+            List<Tuple<string, string, string, string, string, string>> odfpList =
+                new List<Tuple<string, string, string, string, string, string>>();
+            try
+            {
+                List<string> dgList = new DesignGroups().GetDesignGroups(connSource, queryDesignGroups);
+                List<DesignFilePath> idfpList = new InputDesignFilePaths().GetInputDesignFilePaths(XtplDir, dgList);
+                
+                // DesignID, SchoolID, DesignGroup, Selector
+                List<Tuple<string, string, string, string>> dList = 
+                    new Designs().GetDesigns(connSource, queryDesigns);
+                string rpt = $"DesignGroup Count: {dgList.Count}  Designs Count: {dList.Count}"
+                    + $"  Xtpl Count: {idfpList.Count}";
+                file.WriteLine(rpt);
+                file.WriteLine();
+                file.Flush();
+                logger.Info(rpt);
+                odfpList = new SelectorProcessor().GetDesignPaths(idfpList, dgList, dList);
+            }
+            catch (Exception ex)
```

Seems like more than one exception type should be used here - IOException, whatever GetDesignPaths throws, then a generic Exception.

---

See my previous comments on C# and .Net exceptions, most methods through more than one type of exception and since I am logging the exception and that contains the exception type and the error information, the catch all exception syntax seems the best approach here.